### PR TITLE
Document MySQL row limit issue with Vitess

### PR DESCRIPTION
**Pull request by abdhabli**
Check this Slack thread [here](https://metaboat.slack.com/archives/C864UT5CZ/p1728490676071639).
Add a section about a known issue with Vitess and MySQL row limits.

### CODE REVIEW

File: docs/databases/connections/mysql.md [hunk 1]
```diff
@@ -172,6 +172,10 @@ mysql:
   command: ["--default-authentication-plugin=mysql_native_password"]
 ```
 
+## Limitations with Vitess-based databases
```
Comment (jeff-bruemmer) [inline docs/databases/connections/mysql.md:175]:
```suggestion
## When querying Vitess, you should limit your subqueries
```

File: docs/databases/connections/mysql.md [hunk 2]
```diff
@@ -172,6 +172,10 @@ mysql:
...
+
+When using Metabase with MySQL databases which use Vitess as middleware, there can be issues with how row limits are handled. Typically, Metabase applies limits (e.g., 2000 or 10000 rows) to the final query results. However, due to a known bug in Vitess, these limits might be applied to subqueries instead, leading to unexpected results. To avoid this behavior, add a `LIMIT` clause inside every subquery manually to control the number of rows returned.
```
Comment (jeff-bruemmer) [inline docs/databases/connections/mysql.md:177]:
```suggestion
When querying Vitess databases, you should add a `LIMIT` clause inside each subquery.

The reason: typically, Metabase applies limits (e.g., 2000 or 10000 rows) to the final query results. But due to a known bug in Vitess, Vitess might apply these limits to subqueries, which can lead to unexpected results. The workaround is to add limits to each of your subqueries.
```

Does this clarify it? I wanted to lead with what people should do, then explain why.

**Review by jeff-bruemmer — APPROVED** [review]
Approved with comments to address before merging.

Thanks, Abed.

**Comment by alexyarosh** [comment]
@abdhabli don't forget to backport it to 50 as well once you merge 😉

**Review by Tony-metabase — APPROVED** [review]

**Pull request merged**